### PR TITLE
npm: support ^= and ~= grammar

### PR DESCRIPTION
**Pull request by zllak**
## Issue (#22)
When parsing `package.json` files on public repos, I stumbled upon the `^=` syntax.
After looking at the node semver implementation, as far as version 7.6.2, it seems to be totally legal.
Unfortunately this behavior is not documented in the official node semver grammar.

`^=1.0.0` is equivalent to `^1.0.0`
`~=1.0.0` is equivalent to `~1.0.0`

Semver version:
```semver@7.6.2```
```
Welcome to Node.js v22.3.0.
Type ".help" for more information.
> const semver = require('semver')
> semver.validRange('~=1.0.0')
'>=1.0.0 <1.1.0-0'
> semver.validRange('~===1.0.0')
'>=1.0.0 <1.1.0-0'
> semver.validRange('^=1.0.0')
'>=1.0.0 <2.0.0-0'
> semver.validRange('^===1.0.0')
'>=1.0.0 <2.0.0-0'
> semver.validRange('~>=1.0.0')
'>=1.0.0 <1.1.0-0'
> semver.validRange('~>===1.0.0')
'>=1.0.0 <1.1.0-0'
```

## Fix
In this PR, I propose a fix where I simply skip `=` for tilde and caret, as they don't impact the way the partial is treated.

This fixes issue #22

### CODE REVIEW

File: src/range.rs [hunk 1]
```diff
@@ -330,17 +330,12 @@ impl Partial {
       XRange::Val(major) => match self.minor {
         XRange::Wildcard => Version {
           major: major + 1,
-          minor: 0,
-          patch: 0,
-          pre: Vec::new(),
-          build: Vec::new(),
+          ..Default::default()
```
Comment (dsherret) [inline src/range.rs:333]:
```suggestion
          minor: 0,
          patch: 0,
          pre: Vec::new(),
          build: Vec::new(),
```

Using `..Default::default()` often creates bugs. I'd prefer to keep this explicit so this code is re-evaluated whenever the properties change even though this is unlikely to ever change.

File: src/range.rs [hunk 2]
```diff
@@ -330,17 +330,12 @@ impl Partial {
...
         },
         XRange::Val(minor) => Version {
           major,
           minor: minor + 1,
-          patch: 0,
-          pre: Vec::new(),
-          build: Vec::new(),
+          ..Default::default()
```
Comment (dsherret) [inline src/range.rs:338]:
```suggestion
          patch: 0,
          pre: Vec::new(),
          build: Vec::new(),
```

Same.

**Review by dsherret — APPROVED** [review]
LGTM. Thanks!

**Comment by zllak** [comment]
Thank you !

**Pull request merged**